### DISCUSSION TITLE
How to unit test videojs record feature with Karma-Jasmine in Angular?

### DISCUSSION THREAD

**Discussion by JunathAhmed** [videojs/video.js#8379] [Q&A]
### Description

I'm trying to unit test start record in Video.js. I have created a mock player object for video.js. When I unit test it, it breaks.I need code coverage 100% for videojs in angular

### Reduced test case

https://www.example.com

### Steps to reproduce
// code

     authBtn() { 
        if(this.deviceId ==''){
          this.helper.toastErrorMessage('Please Select Camera');
          return;
        }
        playerStatus = true;
        player.record().start();
      }

// Unit Test Code

      describe('AuthenticateCandidateComponent', () => {
      it('should display error message and return if deviceId is empty', () => {
        spyOn(component.helper, 'toastErrorMessage');
        component.deviceId = ''; 
        component.playerStatus=true;
        component.authBtn(); expect(component.helper.toastErrorMessage).toHaveBeenCalledWith('Please Select Camera');
        expect(component.playerStatus).toBeTrue();
      });
    
      it('should start recording when authBtn is called', () => {
        component.player = {
          record: jasmine.createSpyObj('record', ['start']), 
        };
        component.authBtn();
        expect(component.playerStatus).toBe(true);
        expect(component.player.record().start).toHaveBeenCalled(); // Unit Test breaks here 
      });
      
      });

### Errors

```
TypeError: Cannot read properties of undefined (reading 'record')
    at AuthenticateCandidateComponent.selectedData (http://localhost:9876/_karma_webpack_/webpack:/src/app/views/proctoring/authenticate-candidate/authenticate-candidate.component.ts:221:5)
    at UserContext.<anonymous> (http://localhost:9876/_karma_webpack_/webpack:/src/app/views/proctoring/authenticate-candidate/authenticate-candidate.component.spec.ts:369:15
```

### What version of Video.js are you using?

 Video.js 7.6.6

### Video.js plugins used.

yes

### What browser(s) including version(s) does this occur with?

Version 115.0.5790.98

### What OS(es) and version(s) does this occur with?

ubuntu

[a comment by an automated account is not shown]

**Comment by welcome** [comment]
👋 Thanks for opening your first issue here! 👋

If you're reporting a 🐞 bug, please make sure you include steps to reproduce it. We get a lot of issues on this repo, so please be patient and we will get back to you as soon as we can.
To help make it easier for us to investigate your issue, please follow the [contributing guidelines](https://github.com/videojs/video.js/blob/master/CONTRIBUTING.md).

**Comment by mister-ben** [comment]
`component.player` is undefined... you probably want to be doing something like `component.player = videojs('xyz')` when setting up the test
